src/main/java/controller/BusController.java

```diff
@@ -1,28 +1,94 @@
-
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import entity.Bus;
+import java.sql.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 
 public class BusController {
-	DatabaseHandler databaseHandler = new DatabaseHandler();
-	Statement statement = databaseHandler.createStatement();
 
-	public void listAllBuses() {
+	public int listAllBuses() {
+		DatabaseHandler databaseHandler = new DatabaseHandler();
+		Statement statement = databaseHandler.createStatement();
+		int id = -1;
+		System.out.println("List of all buses:");
 		try {
+			if (databaseHandler.getDbConnection() != null) {
 			ResultSet resultSetBuses = statement.executeQuery("SELECT * FROM buses");
 			while (resultSetBuses.next()) {
-				int myDriverId = resultSetBuses.getInt("id");
-				String myBusNumber = resultSetBuses.getString("busNumber");
-				float myBusFuel = resultSetBuses.getFloat("fuel");
-				System.out.println(myDriverId + " " + myBusNumber + " " + myBusFuel);
+				id = resultSetBuses.getInt("id");
+				String busNumber = resultSetBuses.getString("busNumber");
+				int driverId = resultSetBuses.getInt("driverId");
+				float fuel = resultSetBuses.getFloat("fuel");
+				Timestamp createdOn = resultSetBuses.getTimestamp("createdOn");
+				Date purchasedOn = resultSetBuses.getDate("purchasedOn");
+
+				Bus newBus = new Bus(id, busNumber, driverId, fuel, createdOn, purchasedOn);
+
+				System.out.println(newBus.toString());
 			}
+			statement.close();
 			databaseHandler.closeConnection();
-
+			}
+			return id;
 		} catch (SQLException ex) {
-			System.out.println(ex);
+			ex.printStackTrace();
+			return id;
+		}
+	}
+
+	public int findBusByNumber(String busNumber) {
+		DatabaseHandler databaseHandler = new DatabaseHandler();
+		Statement statement = databaseHandler.createStatement();
+		int id = -1;
+		try {
+			if (databaseHandler.getDbConnection() != null) {
+			ResultSet resultSetBuses = statement.executeQuery(
+					"SELECT * FROM buses where busNumber = '" + busNumber + "' limit 1");
+			boolean next = resultSetBuses.next();
+				if (next) {
+					id = resultSetBuses.getInt("id");
+					busNumber = resultSetBuses.getString("busNumber");
+					int driverId = resultSetBuses.getInt("driverId");
+					float fuel = resultSetBuses.getFloat("fuel");
+					Timestamp createdOn = resultSetBuses.getTimestamp("createdOn");
+					Date purchasedOn = resultSetBuses.getDate("purchasedOn");
+					Bus newBus = new Bus(id, busNumber, driverId, fuel, createdOn, purchasedOn);
+
+					System.out.println("Bus with id: " + id);
+					System.out.println(newBus.toString());
+				}
+			}
+			return id;
+		} catch (Exception ex) {
+			ex.printStackTrace();
+			return id;
+		}
+	}
+
+	public void create(String busNumber, int driverId, float fuel, String purchasedOn) {
+		DatabaseHandler databaseHandler = new DatabaseHandler();
+		Statement statement = databaseHandler.createStatement();
+
+		if (statement != null) {
```

Comment:
do we need to add that for each statement?

Reply:
To be honest, I do not know if it is a good practice. I found this solution in our **SDA code examples in GitLab** and wanted to try this out